### PR TITLE
bugfix: fix the improper usage of `version` package

### PR DESCRIPTION
**Pull request by mars1024**
Signed-off-by: Bruce Ma <brucema19901024@gmail.com>

#### What type of PR is this?
/bugfix

#### What this PR does / why we need it:
fix the improper usage of `version` package, which is causing unexpected results in version comparison.

#### Which issue(s) this PR fixes:

NONE

### CODE REVIEW

File: pkg/utils/version.go [hunk 1]
```diff
@@ -16,15 +16,16 @@ limitations under the License.
...
-	KubeV1240 = "v1.24.0"
+var (
+	// KubeV1240Alpha4 indicates the version of Kubernetes v1.24.0-alpha.4
+	KubeV1240Alpha4 = version.MustParseSemantic("v1.24.0-alpha.4")
 )
 
 // SATokenAutoGenerated is used to indicate whether secret-based service account token can be auto generated
 func SATokenAutoGenerated(kubeServerVersion string) bool {
-	return version.CompareKubeAwareVersionStrings(kubeServerVersion, KubeV1240) > 0
+	return version.MustParseSemantic(kubeServerVersion).LessThan(KubeV1240Alpha4)
```
Comment (dixudx) [inline pkg/utils/version.go:30]: `MustParseSemantic` will cause a panic error. Better wrap with a closure or handle the returned error.

File: pkg/utils/version_test.go [hunk 2]
```diff
@@ -40,11 +40,19 @@ func TestSATokenAutoGenerated(t *testing.T) {
 			want:              false,
 		},
 		{
-			kubeServerVersion: "v1.24.6+k3s1",
```
Comment (dixudx) [inline pkg/utils/version_test.go:43]: This is the version of a k3s release.

Comment (mars1024) [reply pkg/utils/version_test.go:43]: Really? Do you mind running `kubectl version` in k3s cluster and print the output here?

Comment (mars1024) [reply pkg/utils/version_test.go:43]: Actually, I removed this case because it failed.

Comment (mars1024) [reply pkg/utils/version_test.go:43]: image

Comment (dixudx) [reply pkg/utils/version_test.go:43]:
```bash
$ k3s kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3+k3s1", GitCommit:"f2585c1671b31b4b34bddbb3bf4e7d69662b0821", GitTreeState:"clean", BuildDate:"2022-10-25T19:59:38Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
```

Comment (mars1024) [reply pkg/utils/version_test.go:43]: A little bit confused, if [https://github.com/k3s-io/kubernetes](https://github.com/k3s-io/kubernetes) only have tags named `v*-k3s1`, then where `v1.25.3+k3s1` come from?
Also, in your output, only have client version, what about the server version?

Comment (mars1024) [reply pkg/utils/version_test.go:43]: Updated.

File: pkg/utils/version.go (hunk 1 above)
Comment (mars1024) [reply pkg/utils/version.go]: Updated.

**Review by dixudx — APPROVED** [review]
Nice work!
Welcome to Clusternet community.

**Pull request merged**